### PR TITLE
BasicDateTimeTestSuite reworked

### LINKED ISSUE #113: Refactor: Rework BasicDateTimeTestSuite to be more in-line with newer tests
### Description

BasicDateTimeTestSuite contains 9 facts at the moment, with multiple asserts per fact. This should be broken up into more facts to be more thorough.

### Script Name

BasicDateTimeTestSuite 

### Steps To The Task

- [x] Split up all tests within existing facts into their own facts
- [x] Add comments
- [x] Update assert descriptions

### PR DESCRIPTION
**Pull request by AdamBorek**
Facts have been broken up into multiple smaller ones

### CODE REVIEW

File: projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml [hunk 1]
```diff
@@ -1,216 +1,494 @@
...
+	
+	addFact("date_inc_test #7", function() {
+			
+		var output, input = date_create_datetime(2011, 9, 15, 9, 43, 44);
 			
 		//#2.2 date_inc_month( 11 )
 		output = date_inc_month(input, 11);
-		assert_equals(date_get_month(output), 8, "#2.2 date_inc_month( 11 ) overflow number of months");
+		assert_equals(date_get_month(output), 8, "date_inc_month( 11 ) overflow number of months");
 		assert_equals(date_get_year(output), date_get_year(input) + 1, "#2.2.1 date_inc_month( 11 ), should go to next year");
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml:293]:
```suggestion

```

File: projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml [hunk 2]
```diff
@@ -1,216 +1,494 @@
...
+	addFact("date_inc_test #13", function() {
+			
+		var output, input = date_create_datetime(2011, 9, 15, 9, 43, 44);
 			
 		//#4.2 date_inc_hour( 30 )
 		output = date_inc_hour(input, 30);
-		assert_equals(date_get_hour(output), 15, "#4.2 date_inc_hour( 30 ) overflow number of hours");
-		assert_equals(date_get_day(output), date_get_day(input) + 1, "#4.2.1 date_inc_hour( 30 ), should go to next day");
+		assert_equals(date_get_hour(output), 15, "date_inc_hour( 30 ) overflow number of hours");
+		assert_equals(date_get_day(output), date_get_day(input) + 1, "date_inc_hour( 30 ), should go to next day");
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml:354]:
```suggestion

```

File: projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml [hunk 3]
```diff
@@ -1,216 +1,494 @@
...
+	addFact("date_inc_test #17", function() {
+			
+		var output, input = date_create_datetime(2011, 9, 15, 9, 43, 44);
 			
 		//#5.2 date_get_minute( 17 )
 		output = date_inc_minute(input, 17);
-		assert_equals(date_get_minute(output), 0, "#5.2 date_inc_minute( 17 ) overflow number of nimutes");
-		assert_equals(date_get_hour(output), date_get_hour(input) + 1, "#5.2.1 date_inc_minute( 17 ), should go to next hour");
+		assert_equals(date_get_minute(output), 0, "date_inc_minute( 17 ) overflow number of nimutes");
+		assert_equals(date_get_hour(output), date_get_hour(input) + 1, "date_inc_minute( 17 ), should go to next hour");
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml:395]:
```suggestion

```

File: projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml [hunk 4]
```diff
@@ -224,19 +502,49 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
...
-
-	addFact("date_span_test", function() {
+	});
+	
+	// DATE SPAN TESTS
+	
+	addFact("date_span_test #1", function() {
 
 		// Set an explicit epsilon to account for different default epsilons in YYC and VM.
 		math_set_epsilon(0.00001)
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml]:
```suggestion
		math_set_epsilon(0.00001);
```

File: projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml [hunk 5]
```diff
@@ -246,95 +554,253 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
...
 		output = date_year_span(firstDateTime, secondDateTime);
-		assert_equals(output, 3.79281352194083, "#1 date_year_span returned the wrong year span between the datetimes");
+		assert_equals(output, 3.79281352194083, "date_year_span returned the wrong year span between the datetimes");
+			
+	});
+	
+	addFact("date_span_test #2", function() {
+
+		// Set an explicit epsilon to account for different default epsilons in YYC and VM.
+		math_set_epsilon(0.00001)
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml]:
```suggestion
		math_set_epsilon(0.00001);
```

File: projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml [hunk 6]
```diff
@@ -246,95 +554,253 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
...
 		output = date_month_span(firstDateTime, secondDateTime);
-		assert_equals(output, 45.51376226329, "#2 date_month_span returned the wrong month span between the datetimes");
+		assert_equals(output, 45.51376226329, "date_month_span returned the wrong month span between the datetimes");
+			
+	});
+	
+	addFact("date_span_test #3", function() {
+
+		// Set an explicit epsilon to account for different default epsilons in YYC and VM.
+		math_set_epsilon(0.00001)
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml]:
```suggestion
		math_set_epsilon(0.00001);
```

File: projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml [hunk 7]
```diff
@@ -246,95 +554,253 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
...
 		output = date_week_span(firstDateTime, secondDateTime);
-		assert_equals(output, 197.903591269841, "#3 date_week_span returned the wrong week span between the datetimes");
+		assert_equals(output, 197.903591269841, "date_week_span returned the wrong week span between the datetimes");
+			
+	});
+	
+	addFact("date_span_test #4", function() {
+
+		// Set an explicit epsilon to account for different default epsilons in YYC and VM.
+		math_set_epsilon(0.00001)
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml]:
```suggestion
		math_set_epsilon(0.00001);
```

File: projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml [hunk 8]
```diff
@@ -246,95 +554,253 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
...
 		output = date_day_span(firstDateTime, secondDateTime);
-		assert_equals(output, 1385.32513888889, "#4 date_day_span returned the wrong day span between the datetimes");
+		assert_equals(output, 1385.32513888889, "date_day_span returned the wrong day span between the datetimes");
+			
+	});
+	
+	addFact("date_span_test #5", function() {
+
+		// Set an explicit epsilon to account for different default epsilons in YYC and VM.
+		math_set_epsilon(0.00001)
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml]:
```suggestion
		math_set_epsilon(0.00001);
```

File: projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml [hunk 9]
```diff
@@ -246,95 +554,253 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
...
 		output = date_hour_span(firstDateTime, secondDateTime);
-		assert_equals(output, 33247.8033333333, "#5 date_hour_span returned the wrong hour span between the datetimes");
+		assert_equals(output, 33247.8033333333, "date_hour_span returned the wrong hour span between the datetimes");
+			
+	});
+
+	addFact("date_span_test #6", function() {
+
+		// Set an explicit epsilon to account for different default epsilons in YYC and VM.
+		math_set_epsilon(0.00001)
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml]:
```suggestion
		math_set_epsilon(0.00001);
```

File: projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml [hunk 10]
```diff
@@ -246,95 +554,253 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
...
 			
-		output = date_second_span(firstDateTime, secondDateTime);
-		assert_equals(output, 119692092, "#7 date_second_span returned the wrong second span between the datetimes");
+	});
 
-	})
+	addFact("date_span_test #7", function() {
+
+		// Set an explicit epsilon to account for different default epsilons in YYC and VM.
+		math_set_epsilon(0.00001)
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml]:
```suggestion
		math_set_epsilon(0.00001);
```

File: projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml [hunk 11]
```diff
@@ -246,95 +554,253 @@ function BasicDateTimeTestSuite() : TestSuite() constructor {
...
+	});
+	
+	// DATE TIMEZONE TESTS
+	
+	addFact("date_timezone_test #1", function() {
 
 		var old = date_get_timezone();
 
 		date_set_timezone(timezone_local);
 		assert_equals(date_get_timezone(), timezone_local, "#1 date_set_timezone( timezone_local ), failed to switch timezones");
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDateTimeTestSuite/BasicDateTimeTestSuite.gml]:
```suggestion
		assert_equals(date_get_timezone(), timezone_local, "date_set_timezone( timezone_local ), failed to switch timezones");
```

**Review by Indibulous — APPROVED** [review]
Seems good! Added a few semicolons and fixed a typo - also marked a few lines that i think shouldn't be there? They look like they have been copied into the test below but forgot to be removed from the tests they are in.

Also, not super important, but might be worth removing all the comments with the old test numbers.

**Comment by DiasFranciscoA** [comment]
@AdamBorek probably better to clean the old comments out of them to avoid misleading.. on of the reasons for the descriptions in the tests is so that you can do a CTRL + SHIFT + F and find the place in the code where the assert failed.. for easier debugging. so removing unnecessary comments is good. Also when inside loops try to make sure the loop index is some where in the description of the assert this way it's possible to know what iterations of the loop is causing the issue 🤔

**Pull request merged**